src/com/owncloud/android/ui/activity/PatternLockActivity.java

```diff
@@ -0,0 +1,275 @@
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.ui.activity;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.andrognito.patternlockview.PatternLockView;
+import com.andrognito.patternlockview.listener.PatternLockViewListener;
+import com.andrognito.patternlockview.utils.PatternLockUtils;
+import com.owncloud.android.BuildConfig;
+import com.owncloud.android.R;
+import com.owncloud.android.lib.common.utils.Log_OC;
+
+import java.util.List;
+
+public class PatternLockActivity extends AppCompatActivity {
+
+    private static final String TAG = PatternLockActivity.class.getSimpleName();
+
+    public final static String PREFERENCE_SET_PATTERN = "set_pattern";
+    public final static String ACTION_REQUEST_WITH_RESULT = "ACTION_REQUEST_WITH_RESULT";
+    public final static String ACTION_CHECK_WITH_RESULT = "ACTION_CHECK_WITH_RESULT";
+    public final static String ACTION_CHECK = "ACTION_CHECK_PATTERN";
+
+    public final static String KEY_PATTERN = "KEY_PATTERN";
+    public final static String KEY_CHECK_RESULT = "KEY_CHECK_PATTERN_RESULT";
+
+    private static String KEY_CONFIRMING_PATTERN = "CONFIRMING_PATTERN";
+    private static String KEY_PATTERN_STRING = "PATTERN_STRING";
+    private static String PATTERN_HEADER_VIEW_TEXT = "PATTERN_HEADER_VIEW_TEXT";
+    private static String PATTERN_EXP_VIEW_STATE = "PATTERN_EXP_VIEW_STATE";
+    private static String COUNT_VALUE = "COUNT_VALUE";
+
+    private boolean patternPresent = false;
+    private boolean patternExpViewState = false;
+    private String patternValue;
+    private String newPatternValue;
+    private String patternHeaderViewText;
+    private int count = 0;
+
+    private TextView patternHeaderView;
+    private TextView patternExpView;
+    private Button cancelButton;
+    private PatternLockView patternLockView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (!BuildConfig.DEBUG) {
+            getWindow().addFlags(WindowManager.LayoutParams.FLAG_SECURE);
+        }
+        setContentView(R.layout.activity_pattern_lock);
+        patternHeaderView = (TextView) findViewById(R.id.header_pattern);
+        patternExpView = (TextView) findViewById(R.id.explanation_pattern);
+        cancelButton = (Button) findViewById(R.id.cancel_pattern);
+        patternLockView = (PatternLockView) findViewById(R.id.pattern_lock_view);
+        patternLockView.clearPattern();
+        if (ACTION_CHECK.equals(getIntent().getAction())) {
```

Comment:
I miss some comments here to explain the different actions in this activity. I guess this activity is being used to request the pattern the first time user enables the feature and to check if is correct when opening the app. Can you include a comment for each if else case explaining a bit that?